src/renderer/components/ReactFlowBox.tsx

```diff
@@ -135,19 +148,69 @@ const updateZIndexes = (
         }
     }
 
+    const nodesToAdjust: { node: Node<NodeData>; zIndex: number }[] = [];
     // set the zIndex of all edges
     for (const e of edges) {
         let zIndex = Math.max(
             byId.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
             byId.get(e.target)?.zIndex ?? STARTING_Z_INDEX
         );
 
+        const connectedNodes = nodes.filter((n) => n.id === e.source || n.id === e.target);
```

Comment:
When the user moves nodes, this code runs every frame, so it needs to be fast. This is why I made sure the whole zIndex code runs in O(n) and is as fast as I could make it.

While I don't want you to benchmark everything (not that it would even be that useful), please write your code so that it's still O(n) (here n being the number of nodes + edges). 

`byId.get` could be used in this specific line of code to reduce the time complexity from O(n) to O(1), so please use it like the code above already does.

Reply:
This code only runs on node drag stop, doesnt it? That means once you stop dragging, not every frame.

Reply:
No, on move.

Reply:
Ah, yup